File: packages/android-im-ui/src/test/kotlin/com/flare/im/ui/MessageContentContractTest.kt
package com.flare.im.ui

import org.junit.Assert.assertEquals
import org.junit.Assert.assertNull
import org.junit.Assert.assertTrue
import org.junit.Test

class MessageContentContractTest {
    @Test fun `message content projection covers RC kinds`() {
        assertEquals(26, FlareMessageContentKind.entries.size)
        assertEquals("image_group", resolveMessageContentContract(FlareMessageContentKind.MultiImage).wireType)
        assertNull(resolveMessageContentContract(FlareMessageContentKind.ReadOnce).wireType)
        assertTrue(FlareMessageContentKind.Code.capabilities.contains(FlareMessageContentCapability.HorizontalScroll))
    }
}

File: packages/flutter-im-ui/test/message_content_contract_test.dart
import 'package:flare_im_ui/flare_im_ui.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  test('message content projection covers RC kinds', () {
    expect(FlareMessageContentKind.values, hasLength(26));
    expect(
      resolveMessageContentContract(
        FlareMessageContentKind.multiImage,
      ).wireType,
      'image_group',
    );
    expect(
      resolveMessageContentContract(FlareMessageContentKind.readOnce).wireType,
      isNull,
    );
    expect(
      resolveMessageContentContract(FlareMessageContentKind.code).capabilities,
      contains(FlareMessageContentCapability.horizontalScroll),
    );
  });
}
